lib/proxy/plugin_validators.rb

```diff
@@ -57,6 +57,21 @@ def validate!(settings)
     end
   end
 
+  class OptionalUrl < Base
+    def validate!(settings)
+      setting_value = settings[@setting_name]
+
+      return true if setting_value.nil?
+
+      parsed = URI.parse(setting_value)
+      raise ::Proxy::Error::ConfigurationError, "Setting '#{@setting_name}' is missing a scheme" if parsed.scheme.nil? || parsed.scheme.empty?
+
+      true
+    rescue URI::InvalidURIError
+      raise ::Proxy::Error::ConfigurationError.new("Setting '#{@setting_name}' contains an invalid url")
+    end
+  end
```

Comment:
I'd suggest this should inherit from `Url`
```suggestion
  class OptionalUrl < Url
    def validate!(settings)
      settings[@setting_name].nil? || super
    end
  end
```